0000-remove-static-assert.md

```diff
@@ -0,0 +1,73 @@
+- Feature Name: remove-static-assert
+- Start Date: 2015-04-28        
+- RFC PR: 
+- Rust Issue: https://github.com/rust-lang/rust/pull/24910
+
+# Summary
+
+Remove the `static_assert` feature.
+
+# Motivation
+
+To recap, `static_assert` looks like this:
+
+```rust
+#![feature(static_assert)]
+#[static_assert]
+static asssertion: bool = true;
+```
+
+If `assertion` is `false` instead, this fails to compile:
+
+```text
+error: static assertion failed
+static asssertion: bool = false;
+                          ^~~~~
+```
+
+If you don’t have the `feature` flag, you get another interesting error:
+
+```text
+error: `#[static_assert]` is an experimental feature, and has a poor API
+```
+
+Throughout its life, `static_assert` has been... weird. Graydon suggested it
+[in May of 2013][suggest], and it was
+[implemented][https://github.com/rust-lang/rust/pull/6670] shortly after.
+[Another issue][issue] was created to give it a ‘better interface’. Here’s why:
+
+> The biggest problem with it is you need a static variable with a name, that
+> goes through trans and ends up in the object file.
+
+In other words, `assertion` above ends up as a symbol in the final output. Not
+something you’d usually expect from some kind of static assertion.
+
+[suggest]: https://github.com/rust-lang/rust/issues/6568
+[issue]: https://github.com/rust-lang/rust/issues/6676
+
+So why not improve `static_assert`? With compile time function evaluation, the
+idea of a ‘static assertion’ doesn’t need to have language semantics. Either
```

Comment:
I suppose with CTFE, this `static_assert!()` macro could be provided by a crate, although it seems like a reasonable candidate to bake into the language (e.g. because the standard libraries might want to use it).